backend/tracim_backend/lib/core/content.py

```diff
@@ -536,29 +541,34 @@ def create(self, content_type_slug: str, workspace: Workspace, parent: Content=N
                         content_id=workspace.workspace_id,
                     )
                 )
-        if filename:
-            label = os.path.splitext(filename)[0]
-        if label:
-            self._is_content_label_free_or_raise(
-                label,
+        content = None
+        if filename or label:
+            if label:
```

Comment:
building filename should not be implemented here. you should have a method somewhere returning the filename of a content (from the content property or from a dynamically build one).

Reply:
all these lines should be written like:

```
filename = content.get_filename()  # this encapsulate the method to build dynamically the filename
if filename:
    self._is_filename_free_or_raise(
                filename,
                workspace,
                parent,
            )
```

Reply:
For clarify, i prefer to do check before creating/setting content object, but in this case, that not easy.

one solution is to begin creating object with label and filename then check correctness of filename then raise error which should rollback object creation.

Reply:
so a solution would be to move filename build operation into a method of ContentApi called for example `prepare_filename(x, y)`.

Th eway to create the filename (from filename property or from label + type of content) has not to be visible in the current algorithm

Reply:
Done.